distribution/lib/Standard/Table/0.0.0-dev/src/Excel/Excel_Workbook.enso

```diff
@@ -0,0 +1,140 @@
+from Standard.Base import all
+import Standard.Base.Error.File_Error.File_Error
+import Standard.Base.Error.Illegal_Argument.Illegal_Argument
+from Standard.Base.Data.Filter_Condition import sql_like_to_regex
+
+from Standard.Base.Metadata.Widget import Single_Choice
+from Standard.Base.Metadata.Choice import Option
+import Standard.Base.Metadata.Display
+
+import project.Data.Table.Table
+import project.Excel.Excel_Range.Excel_Range
+import project.Excel.Excel_Reader
+
+polyglot java import org.enso.table.read.ExcelReader
+polyglot java import org.apache.poi.ss.usermodel.Workbook
+
+type Excel_Workbook
+    ## Load a File as an Excel_Workbook.
+
+       Arguments:
+       - file: The file to load.
+       - xls_format: Whether to use the old XLS format (default is XLSX).
+       - headers: Whether to use the first row as headers (default is to infer).
+    new : File | Text -> Boolean -> Boolean | Infer -> Excel_Workbook
+    new file xls_format=False headers=Infer =
+        file_obj = File.new file . normalize
+        File_Error.handle_java_exceptions file_obj <|
+            file_obj.with_input_stream [File_Access.Read] stream->
+                stream.with_java_stream java_stream->
+                    workbook = ExcelReader.readWorkbook java_stream xls_format
+                    Excel_Workbook.Value workbook file_obj xls_format headers
+
+    ## PRIVATE
+       Creates an Excel_Workbook connection.
+    Value workbook:Workbook file:File xls_format:Boolean headers:(Boolean|Infer)
+
+    ## Returns the list of databases (or catalogs) for the connection.
+    databases : Nothing
+    databases self = Nothing
+
+    ## Returns the name of the current database (or catalog).
+    database : Text
+    database self = self.file.path
+
+    ## Returns a new Connection with the specified database set as default.
+
+       Arguments:
+        - database: The name of the database to connect to.
+    @database (Single_Choice display=Display.Always values=[Option 'Nothing'])
+    set_database : Text -> Excel_Workbook ! Illegal_Argument
+    set_database self database =
+        if database == self.database then self else
+            file = File.new database
+            if file.exists && file.is_directory.not then Excel_Workbook.new file self.xls_format self.headers else
```

Comment:
What if the switched-to file does not hav the same `xls_format` as `self`? Shouldn't we rather infer it?

If `set_database` works for adjacent files, shouldn't `databases` return those?

Reply:
I will think more on this one as we work through the `Folder` connection idea. At which point I want to revisit this bit.